spec/DBClientSpec.js

```diff
@@ -0,0 +1,253 @@
+require('dotenv').config();
+const Cloudant = require('@cloudant/cloudant');
+ const jasmine = require('jasmine');
+const DBClient = require('../api/lib/DBClient');
+const JSONValidator = require('../api/lib/JSONValidator');
+ let jasmineTimeout = 10000;
+ describe('DBClient tests setup', () => {
+    beforeAll(() => {
+        const user = process.env.CLOUDANT_USER;
+        const pw = process.env.CLOUDANT_PW;
+        this._client = Cloudant({ account: user, password: pw });
+    });
+     beforeEach((done) => {
+        const menuEntry = {
+            "_id": "test",
+            "restaurant": {
+                "name": "test",
+                "address": "test",
+                "city": "test"
+            },
+            "file": "test",
+            "ocr": []
+        }
+        this._client.db.create('menus-test', (err) => {
+            if (err) return done(err);
+            const database = this._client.db.use('menus-test');
+            database.insert(menuEntry, (err, data) => {
+                if (err) return done(err);
+                return done();
+            });
+        });
+    }, jasmineTimeout);
+     afterEach((done) => {
+        this._client.db.destroy('menus-test', (err, data) => {
+            if (err) return done(err);
+            return done();
+        });
+    }, jasmineTimeout);
+     describe('DBClient constructor', () => {
+        it('should throws an error when db param is different than undefined/string', () => {
+            expect(function () { new DBClient(1); })
+                .toThrowError('Wrong type for parameter db. Received number, expected string');
+        });
+         it('should have a null value for _db when db param is empty', () => {
+            let dbInstance = new DBClient();
+            expect(dbInstance._db).toBeNull();
+        });
+         it('should have a string value for _db when db param is string', () => {
+            let isString = true;
+            let dbInstance = new DBClient('menus-test');
+            if (typeof dbInstance._db !== 'string') isString = false;
+            expect(isString).toBe(true);
```

Comment:
in this case you could also use `toBeTruthy()`, just keep in mind that `toBeTruthy()` is more generic than `toBe(value)`, because things like "a", 1.0, 42, etc are considered "Truthy".

Reply:
This is not a blocker though, just a comment

Reply:
OK, thanks! :+1: